From 559191bbdae3cae1eeca62f093a60d7379cab96e Mon Sep 17 00:00:00 2001
From: "wei.du" <wei.du@amlogic.com>
Date: Sun, 26 Sep 2021 19:04:29 +0800
Subject: [PATCH 2/2] asink: do scaletempo_update_segment before avsync session
 create [1/1]

PD#SWPL-58535

Problem:
1. auto connect to alsasink
2. not do scaletempo_update_segment when avsync session create failed.

Solution:
1. increase rank for asink
2. do scaletempo_update_segment before avsync session create.

Verify:
AP222

Change-Id: I24bd4f367836f149785c708d49954fd2b59329f7
Signed-off-by: wei.du <wei.du@amlogic.com>
---
 src/gstamlhalasink_new.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/gstamlhalasink_new.c b/src/gstamlhalasink_new.c
index 82959b2..363334a 100644
--- a/src/gstamlhalasink_new.c
+++ b/src/gstamlhalasink_new.c
@@ -1843,6 +1843,9 @@ gst_aml_hal_asink_event (GstAmlHalAsink *sink, GstEvent * event)
         bypass_avs = TRUE;
 #endif
 
+      if (priv->tempo_used)
+        scaletempo_update_segment (&priv->st, &priv->segment);
+
       /* create avsync before rate change */
       if (!priv->avsync && priv->direct_mode_ && !bypass_avs) {
         char setting[20];
@@ -1876,9 +1879,6 @@ gst_aml_hal_asink_event (GstAmlHalAsink *sink, GstEvent * event)
         priv->stream_->common.set_parameters (&priv->stream_->common, setting);
       }
 
-      if (priv->tempo_used)
-        scaletempo_update_segment (&priv->st, &priv->segment);
-
       if (priv->direct_mode_) {
         update_avsync_speed(sink, segment.rate);
         GST_INFO_OBJECT (sink, "rate to %f", segment.rate);
@@ -3661,7 +3661,7 @@ GstClock *gst_aml_hal_asink_get_clock (GstElement *element)
 static gboolean
 plugin_init (GstPlugin * plugin)
 {
-  return gst_element_register (plugin, "amlhalasink", GST_RANK_PRIMARY,
+  return gst_element_register (plugin, "amlhalasink", GST_RANK_PRIMARY + 1,
       GST_TYPE_AML_HAL_ASINK);
 }
 
-- 
2.25.1

